packages/shylu/shylu_dd/frosch/src/InterfaceSets/FROSch_DDInterface_def.hpp

```diff
@@ -473,8 +472,8 @@ namespace FROSch {
                 tmpNodesMap = Xpetra::MapFactory<LO,GO,NO>::Build(NodesMap_->lib(),-1,indices(),0,NodesMap_->getComm());
             }
 
-            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMat = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpNodesMap,10);
-            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMatTmp = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpUniqueNodesMap,10);
+            GraphPtr commGraph = Xpetra::CrsGraphFactory<LO,GO,NO>::Build(tmpNodesMap,10);
```

Comment:
I have a suspicion of what you're trying to do here.  It looks like you're building a "communication graph" to represent the communication pattern.  You don't have to do that.  That's what Import and Export do.

Reply:
I have an ``overlapping map'', and I need to find out which processes share the replicated elements. Therefore, I am using Import and Export.

Reply:
@searhein wrote:
> I have an "overlapping map," and I need to find out which processes share the replicated elements.

I think you could use the two-argument overload of `Tpetra::Map::createOneToOne`, with a custom TieBreak object that remembers the overlap.  See lines 757-834 and 903-926 of `tpetra/core/src/Tpetra_DirectoryImpl_def.hpp` to see how the TieBreak object is used.

Reply:
@mhoemmen I'll have to take a closer look at this. Probably, I would do this in another PR. 

However, just to clarify this for me: using the method createOneToOne and an TieBreak object, I could obtain the list of processes sharing an element. I need to obtain this list on each processes owning the element on the overlapping map. 
Doesn't this always require two communication steps, as in my current implementation?

Reply:
Perhaps.  I haven't thought too much about it.  I'm just presenting it as an alternative.

Reply:
@mhoemmen I will look into this. At least for `Tpetra`, this could really be an alternative.